### ISSUE TITLE
Issue creating /tmp/cmst0/.wmcore_cache file 

### ISSUE THREAD

**Issue by jhonatanamado** [dmwm/WMCore#10800]
**Impact of the bug**
T0 Agent not able to create/run any job/workflow using WMCore agent 1.5.2

**Describe the bug**
T0 use the Tier0Feeder component to create the sandbox. When Tier0 started to test a new release based con WMCore 1.5.2, Tier0Feeder component was unable to start properly given the following error:
```
2021-09-13 06:55:27,603:140108589012736:WARNING:Service:The cachefile /tmp/cmst0/.wmcore_cache_xlpsm5ls/cmssdt.cern.ch/-3895757188456107843_GET__SDT_cgi-bin_ReleasesXML_ does not exi
st and the service at https://cmssdt.cern.ch/SDT/cgi-bin/ReleasesXML/ raised a FileNotFoundError(2, 'No such file or directory') when accessed
2021-09-13 06:55:27,673:140108589012736:ERROR:StdBase:About to raise exception <@========== WMException Start ==========@>
Exception Class: WMSpecFactoryException
Message: [Errno 2] No such file or directory: '/tmp/cmst0/.wmcore_cache_xlpsm5ls/cmssdt.cern.ch/-3895757188456107843_GET__SDT_cgi-bin_ReleasesXML_'
        ClassName : None
        ModuleName : WMCore.WMSpec.WMWorkloadTools
        MethodName : _validateArgFunction
        ClassInstance : None
        FileName : /data/tier0/srv/wmagent/3.0.0/sw.jamadova/slc7_amd64_gcc630/cms/t0/3.0.0/lib/python3.8/site-packages/WMCore/WMSpec/WMWorkloadTools.py
        LineNumber : 105
        ErrorNr : 0
```

**How to reproduce it**
Deploying the T0 (3.0.0) agent and start all the components of the agent or only Tier0Feeder.

**Expected behavior**
The /tmp/cmst0/.wmcore_cache file must be correctly created and finally Tier0Feeder will be able to create the sandbox

**Additional context and error message**
Issue is **only** affecting when T0 use WMCore 1.5.2.

**Comment by amaltaro** [comment]
After many trials and help from the T0 team, I think I know understand what the problem is.

The exact line where things fail is here (when validating the CMSSWVersion):
https://github.com/dmwm/WMCore/blob/67ef9eca98143100ca260a53a3a15a06f8af1d4f/src/python/WMCore/Services/Service.py#L297
and the exception is:
```
Traceback (most recent call last):
  File "/data/tier0/srv/wmagent/3.0.1/sw/slc7_amd64_gcc630/cms/t0/3.0.1/lib/python3.8/site-packages/WMCore/Services/Service.py", line 331, in getData
    with open(cachefile, 'w') as f:
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/cmst0/.wmcore_cache_9f_jrv7h/cmssdt.cern.ch/-7578808970081787174_GET__SDT_cgi-bin_ReleasesXML_'
```

however, that directory to hold the cache file should have been created when `TagCollector` is instantiated, basically following these steps:
https://github.com/dmwm/WMCore/blob/1.5.6/src/python/WMCore/Services/Requests.py#L326
and then this temporary directory (with delete on exit):
https://github.com/dmwm/WMCore/blob/67ef9eca98143100ca260a53a3a15a06f8af1d4f/src/python/WMCore/Services/Requests.py#L369

apparently this cache directory gets deleted before our first cachefile gets written out, causing the FileNotFound exception.

If we want to enforce the usage of a non temporary cache directory (thus not letting it to go away in a random basis), we should define the following variable in the `env.sh` script:
```
export WMCORE_CACHE_DIR=/tmp/cmst0
```
which would not reach the else clause calling `mkdtemp`.

To summarize, I do not think we should change anything in the code. Instead, the T0 team needs to define that new variable in the env.sh script and this problem would not occur again.
